src/mlpack/methods/logistic_regression/logistic_regression_function_impl.hpp

```diff
@@ -70,16 +70,16 @@ double LogisticRegressionFunction<MatType>::Evaluate(
   // multiplied by the squared l2-norm of the parameters then divided by two.
 
   // For the regularization, we ignore the first term, which is the intercept
-  // term.
-  const double regularization = 0.5 * lambda *
-      arma::dot(parameters.col(0).subvec(1, parameters.n_elem - 1),
-                parameters.col(0).subvec(1, parameters.n_elem - 1));
+  // term and take every term except the last one in the decision variable.
+  double norm = arma::norm(parameters.tail_cols(parameters.n_elem - 1));
```

Comment:
I think the use of `tail_cols()` is a nice improvement, but is the switch of `parameters` from `arma::vec` to `arma::rowvec` necessary?  That's an API change and might break some user's code.  I think you could also just do `tail_rows()` and leave it as an `arma::vec`.  Let me know what you think.  Also I think it might be faster (very incrementally) to use `arma::dot()` instead of `arma::norm()` and then multiplying.